packages/aws-cdk/test/api/exec.test.ts

```diff
@@ -0,0 +1,126 @@
+import * as cxschema from '@aws-cdk/cloud-assembly-schema';
+import * as cdk from '@aws-cdk/core';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as semver from 'semver';
+import { ImportMock } from 'ts-mock-imports';
+import { SdkProvider } from '../../lib';
+import { execProgram } from '../../lib/api/cxapp/exec';
+import { Configuration, Settings } from '../../lib/settings';
+
+// We need to increase the default 5s jest
+// timeout for async tests because the 'execProgram' invocation
+// might take a while :\
+const TEN_SECOND_TIMEOUT = 10000;
+
+function createApp(outdir: string): cdk.App {
+
+    const app = new cdk.App({outdir});
+    const stack = new cdk.Stack(app, 'Stack');
+
+    new cdk.CfnResource(stack, "Role", {
+        type: "AWS::IAM::Role",
+        properties: {
+            RoleName: "Role"
+        }
+    });
+
+    return app;
+}
+
+async function createSdkProvider() {
+    return await SdkProvider.withAwsCliCompatibleDefaults();
+}
+
+test('execProgram throws when manifest version > schema version', async () => {
+
+    const outdir = fs.mkdtempSync(path.join(os.tmpdir(), 'exec-tests'));
+
+    const app = createApp(outdir);
+
+    const currentSchemaVersion = cxschema.Manifest.version();
+
+    const mockManifestVersion = semver.inc(currentSchemaVersion, 'major');
```

Comment:
minor comment: should we bump this test to assert that it fails for minor and patch as well?

```suggestion
    const mockManifestVersions[] = [ semver.inc(currentSchemaVersion, 'major'), semver.inc(currentSchemaVersion, 'minor'), semver.inc(currentSchemaVersion, 'patch') ];
```

Reply:
good idea

Reply:
since the actual validation moved to [`manifest.ts`](https://github.com/aws/aws-cdk/blob/epolon/cli-framework-compatibility/packages/@aws-cdk/cloud-assembly-schema/lib/manifest.ts#L54), im adding this as unit tests in the cloud-assembly-schema package. The tests here are now just to ensure the validation actually gets called as part of `execProgram`.

Reply:
See https://github.com/aws/aws-cdk/blob/epolon/cli-framework-compatibility/packages/%40aws-cdk/cloud-assembly-schema/test/manifest.test.ts#L102